File: java/BillsApi/src/main/java/com/apilangarena/javabillsapi/application/bills/create/BillCreatedEvent.java
package com.apilangarena.javabillsapi.application.bills.create;

import java.math.BigDecimal;
import java.time.LocalDate;
import java.time.OffsetDateTime;

public record BillCreatedEvent(
    Long billId,
    String billNumber,
    LocalDate issuedAt,
    BigDecimal subtotal,
    BigDecimal tax,
    BigDecimal total,
    String currency,
    OffsetDateTime occurredAtUtc,
    String source
) {
}

File: kotlin/BillsApi/src/main/kotlin/com/apilangarena/billsapi/application/bills/create/BillCreatedEvent.kt
package com.apilangarena.billsapi.application.bills.create

import java.math.BigDecimal
import java.time.LocalDate
import java.time.OffsetDateTime

data class BillCreatedEvent(
    val billId: Long,
    val billNumber: String,
    val issuedAt: LocalDate,
    val subtotal: BigDecimal,
    val tax: BigDecimal,
    val total: BigDecimal,
    val currency: String,
    val occurredAtUtc: OffsetDateTime,
    val source: String,
)
